### PR TITLE
feat: In-game UI integrated to rest of sample for disconnection/reconnection & player names [MTT-9233] [MTT-9249]

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description
This PR integrates the in-game UI to some of the player/gamplay classes such that the player names are displayed, disconnect/reconnect flow is working. Some gameplay events have been moved into its own asmdef for ease of access & decoupling from some unrelated scripts.

Note: Netcode for GameObjects has been upgraded to v2.1.1.

### Issue Number(s)
[MTT-9233](https://jira.unity3d.com/browse/MTT-9233)
[MTT-9249](https://jira.unity3d.com/browse/MTT-9249)

### Contribution checklist
 - [ N/A ] Tests have been added for the project and/or any internal package
 - [ N/A ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

File: Experimental/DistributedAuthoritySample/Assets/Scripts/GameManagement/GameplayEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+using Unity.Netcode;
+using UnityEngine;
+using System;
+using System.Threading.Tasks;
+
+namespace Unity.Multiplayer.Samples.SocialHub.GameManagement
+{
+    abstract class GameplayEventHandler
```
Comment (hammerlerobi) [inline Experimental/DistributedAuthoritySample/Assets/Scripts/GameManagement/GameplayEventHandler.cs:8]: this should probably be static

**Review by hammerlerobi — APPROVED** [review]
Small things but besides that LGTM

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs [hunk 2]
```diff
@@ -22,16 +24,25 @@ class AvatarTransform : PhysicsObjectMotion, INetworkUpdateSystem
...
+            m_PlayerName.OnValueChanged += OnPlayerNameChanged;
+            OnPlayerNameChanged(string.Empty, m_PlayerName.Value);
+
             if (!HasAuthority)
             {
                 base.OnNetworkSpawn();
                 return;
             }
 
+            m_PlayerName.Value = new FixedString32Bytes(HomeScreenView.PlayerName);
```
[the remark this answers is not in the document]
Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/Assets/Scripts/Player/AvatarTransform.cs]: Good call!

File: Experimental/DistributedAuthoritySample/Assets/Scripts/GameManagement/GameplayEventHandler.cs (hunk 1 above)
Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/Assets/Scripts/GameManagement/GameplayEventHandler.cs]: Ah good catch. I've added this to our MathUtils class too.

**Review by hammerlerobi — APPROVED** [review]

**Pull request merged**